sdks/python/apache_beam/io/textio.py

```diff
@@ -86,7 +90,26 @@ def __init__(self,
                coder,
                buffer_size=DEFAULT_READ_BUFFER_SIZE,
                validate=True,
-               skip_header_lines=0):
+               skip_header_lines=0,
+               header_processor_fns=(None, None)):
+    """Initialize a _TextSource
+
+    Args:
+      header_processor_fns (tuple): a tuple of a header_matcher function
+        and a header_processor function. The header_matcher should
+        return True for all lines at the start of the file that are part
+        of the file header and False otherwise. These header_lines will
+        not be yielded when reading records and instead passed into
+        header_processor to be handled. If skip_header_lines and a
+        header_matcher are both provided, the value of skip_header_lines
```

Comment:
<!--new_thread; commit:5d08ca3beac2fd751bbfdcf83d47e57ce1411bc0; resolved:0-->
nit: use ` for True/False/function names.

Reply:
Done.